Tests/Realm.Tests/Database/RealmValueTests.cs

```diff
@@ -408,6 +408,61 @@ private void RunNumericTests(RealmValue rv, long value, bool isManaged)
             Assert.That(rv != RealmValue.Null);
         }
 
+        [Test]
+        public void EmbeddedObjectAsRealmValue_WhenAddedToRealm_Throws()
```

Comment:
Given that this touches different code paths (`RealmCollectionBase` and `RealmValue`), wouldn't it be better to have a test per code path? If we want to be extreme, even a test per collection with clear title?
The opinion of the other team members is needed here.

Reply:
I agree that dividing them up per collection would make it clearer!